### PR TITLE
pthread: add pthread conditionals tests

### PR DESCRIPTION
**Pull request by lukileczo**
## Description

Added tests for pthread conditionals.

## Motivation and Context

- JIRA: RTOS-179
- Implements tests for https://github.com/phoenix-rtos/libphoenix/pull/186

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested?

- [ ] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [x] Tested by hand on: `ia32-generic-qemu`.

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing linter checks and tests passed.
- [x] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [x] This PR needs additional PRs to work: https://github.com/phoenix-rtos/libphoenix/pull/186
- [ ] I will merge this PR by myself when appropriate.

### CODE REVIEW

**Comment by jsarzy** [comment]
I know is still a draft, but please note that Unity asserts should only be used in the main thread context. It's a macro for `longjmp` therefore when the test fails some weird things can happen. Threads could communicate by shared state and the main thread eventually can verify it using assertions.

**Review by agkaminski — COMMENTED** [review]
General style remark - there should be 2 empty lines between both function declaration and definitions

File: pthread/thread_functions.c [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+	return NULL;
+}
+
+
+void *decrement_count_timed_wait_fail_too_short_timeout(void *args)
+{
+	thread_err_t *thread_err = (thread_err_t *)args;
+	struct timespec time;
+	clock_gettime(CLOCK_REALTIME, &time);
+	time.tv_nsec += 1;
```
Comment (damianloew) [inline pthread/thread_functions.c:72]: Is this needed here?

File: pthread/test_pthread_cond.c [hunk 2]
```diff
@@ -0,0 +1,297 @@
...
+{
+	pthread_condattr_t attr;
+	TEST_ASSERT_EQUAL(EOK, pthread_condattr_init(&attr));
+	TEST_ASSERT_EQUAL(EOK, pthread_condattr_setpshared(&attr, PTHREAD_PROCESS_PRIVATE));
+
+	int pshared;
+	TEST_ASSERT_EQUAL(EOK, pthread_condattr_getpshared(&attr, &pshared));
+	TEST_ASSERT_EQUAL(PTHREAD_PROCESS_PRIVATE, pshared);
+	TEST_ASSERT_EQUAL(PTHREAD_PROCESS_PRIVATE, 1);
+	TEST_ASSERT_EQUAL(-EINVAL, pthread_condattr_setpshared(&attr, PTHREAD_PROCESS_SHARED));
```
[the remark this answers is not in the document]
Comment (lukileczo) [reply pthread/test_pthread_cond.c:64]: From what I've found out our OS only supports `PTHREAD_PROCESS_SHARED`, that's why I've coded it that way.

**Review by agkaminski — APPROVED** [review]
LGTM

**Review by Maxez — APPROVED** [review]
LGTM

**Pull request merged**